xarray/core/array_api_compat.py

```diff
@@ -0,0 +1,41 @@
+import numpy as np
+
+
+def is_weak_scalar_type(t):
+    return isinstance(t, (bool, int, float, complex, str, bytes))
+
+
+def _future_array_api_result_type(*arrays_and_dtypes, xp):
```

Comment:
Add a reference here to https://github.com/data-apis/array-api/issues/805?

Reply:
```suggestion
def _future_array_api_result_type(*arrays_and_dtypes, xp):
    # fallback implementation for `xp.result_type` with python scalars can be removed once a
    # version of the Array API that includes https://github.com/data-apis/array-api/issues/805
    # can be required
```